pkg/server/backuprestoreserver.go

```diff
@@ -291,10 +292,23 @@ func (b *BackupRestoreServer) runServer(ctx context.Context, restoreOpts *brtype
 		},
 	}
 
+	promoteCallback := &brtypes.PromoteLearnerCallback{
+		Promote: func(ctx context.Context, logger *logrus.Entry) {
+			if restoreOpts.OriginalClusterSize > 1 {
+				m := member.NewMemberControl(b.config.EtcdConnectionConfig)
+				if err := m.PromoteMember(ctx); err == nil {
```

Comment:
Should we first check whether the member is sync with the leader before attempting to promote?

Reply:
no, `PromoteMember ` will only succeed if learner is in a sync with a leader of cluster.
So, if it gets failed then it will try again as it is in a leader-elector loop.